packages/salesforcedx-vscode-core/test/vscode-integration/util/cliConfiguration.test.ts

```diff
@@ -125,4 +131,45 @@ describe('SFDX CLI Configuration utility', () => {
       ]);
     });
   });
+
+  describe('ConfigAggregator integration tests', () => {
+    let localSfdxConfigFile: ConfigFile;
+    const dummyLocalDefaultUsername = 'test@local.com';
+
+    afterEach(() => {
+      // Clean up by removing the config file that was created in the setup for this test
+      localSfdxConfigFile.unlinkSync();
+    });
+
+    it.only('Should return the locally configured default username when it exists', async () => {
+      // Arrange: create a local config file with a local default username
+      const localSfdxDirPath = path.join(getRootWorkspacePath(), '.sfdx');
+      process.chdir(localSfdxDirPath);
+
+      class MyConfig extends ConfigFile {
+        public static getFileName(): string {
+          return 'sfdx-config.json';
+        }
+      }
+
+      localSfdxConfigFile = await MyConfig.create({
+        isGlobal: false
+      });
+      localSfdxConfigFile.set(
+        OrgConfigProperties.TARGET_ORG,
+        dummyLocalDefaultUsername
+      );
+      localSfdxConfigFile.writeSync(); // doesn't fire the file watcher logic to run
```

Comment:
interesting...I wonder why not

Reply:
if we used sfdx here to set the default username, would it trigger the filewatcher that would retrigger the reload giving us the correct value below?